api/src/services/script-runner.service.ts

```diff
@@ -605,6 +616,152 @@ export class ScriptRunnerService {
+   */
+  async transferListingAssetData(
+    req: ExpressRequest,
+    dataTransferDTO: AssetTransferDTO,
+    prisma?: PrismaClient,
+  ): Promise<SuccessDTO> {
+    // script runner standard start up
+    const requestingUser = mapTo(User, req['user']);
+    await this.markScriptAsRunStart(
+      `data transfer assets ${dataTransferDTO.jurisdiction}`,
+      requestingUser,
+    );
+
+    // connect to foreign db based on incoming connection string
+    const client =
+      prisma ||
+      new PrismaClient({
+        datasources: {
+          db: {
+            url: dataTransferDTO.connectionString,
+          },
+        },
+      });
+    await client.$connect();
+
+    const doorwayJurisdiction = await this.prisma.jurisdictions.findFirst({
+      where: { name: dataTransferDTO.jurisdiction },
+    });
+
+    if (!doorwayJurisdiction) {
+      throw new Error(
+        `${dataTransferDTO.jurisdiction} county doesn't exist in Doorway database`,
+      );
+    }
+
+    // get jurisdiction
+    const jurisdiction: { id: string }[] =
+      await client.$queryRaw`SELECT id, name FROM jurisdictions WHERE name = ${dataTransferDTO.jurisdiction}`;
+
+    if (!jurisdiction) {
+      throw new Error(
+        `${dataTransferDTO.jurisdiction} county doesn't exist in foreign database`,
+      );
+    }
+
+    const take = calculateTake(40);
+    const skip = calculateSkip(take, dataTransferDTO.page || 1);
+    const listingTransferMap = await this.prisma.listingTransferMap.findMany({
+      take,
+      skip,
+    });
+    console.log(
+      `Found ${listingTransferMap.length} listings on page ${
+        dataTransferDTO.page || 1
+      } out of a possible 40 for this page`,
+    );
+    // loop over each new listing id <-> old listing id relation
+    for (let i = 0; i < listingTransferMap.length; i++) {
+      const oldAssetInfo: {
+        ordinal: number;
+        created_at: Date;
+        updated_at: Date;
+        file_id: string;
+        label: string;
+      }[] = await client.$queryRaw`SELECT
+            li.ordinal,
+            a.created_at,
+            a.updated_at,
+            a.file_id,
+            label
+        FROM listing_images li
+            JOIN assets a ON a.id = li.image_id
+        WHERE li.listing_id = ${listingTransferMap[i].oldId} :: UUID
+          AND a.file_id IS NOT NULL AND a.file_id != ''`;
+      console.log(
+        `moving ${oldAssetInfo.length || 0} assets for listing: ${
+          listingTransferMap[i].oldId
+        }:`,
+      );
+      // loop over each listing image on the old listing
+      for (let j = 0; j < oldAssetInfo.length; j++) {
+        // pull down image from cloudinary
+        const image = await axios.get(
+          `https://res.cloudinary.com/${dataTransferDTO.cloudinaryName}/image/upload/${oldAssetInfo[j].file_id}.jpg`,
+          {
+            responseType: 'arraybuffer',
+          },
+        );
+        const newFileId = (oldAssetInfo[j].file_id as string).replace(
```

Comment:
nit: I don't think this will be an issue, but for some reason it does the path twice. Specifically for this example it is `dev/darwin-dance2_nuowqn/dev/darwin-dance2_nuowqn.jpg` instead of `dev/darwin-dance2_nuowqn.jpg`
```
https://doorway-user-uploads20230329214550248900000003.s3.us-west-1.amazonaws.com/assets/613746/dev/darwin-dance2_nuowqn/dev/darwin-dance2_nuowqn.jpg
```

Reply:
Hey @ludtkemorgan I removed the extra /dev/ but I think the rest of that should be there